lib/improver/tests/utilities/cube_manipulation/test_merge_cubes.py

```diff
@@ -120,18 +138,47 @@ def test_lagged_ukv(self):
     def test_multi_model(self):
         """Test Multi models merge OK"""
         cubes = iris.cube.CubeList([self.cube, self.cube_ukv])
-        result = merge_cubes(cubes)
+        result = merge_cubes(cubes, "mosg__model_configuration")
         self.assertIsInstance(result, Cube)
         self.assertArrayAlmostEqual(
-            result.coord("model_realization").points, [3000., 4000.,
-                                                       4001., 4002.])
+            result.coord("model_realization").points, [0., 1.,
+                                                       2., 1000.])
 
     def test_threshold_data(self):
         """Test threshold data merges OK"""
         cubes = iris.cube.CubeList([self.prob_ukv, self.prob_enuk])
-        result = merge_cubes(cubes)
+        result = merge_cubes(cubes, "mosg__model_configuration")
+        self.assertArrayAlmostEqual(
+            result.coord("model_id").points, [0., 1000.])
+
+    def test_non_mo_model_id(self):
+        """Test that a model ID attribute string can be specified when
+        merging multi model cubes"""
+        cubes = iris.cube.CubeList(
+            [self.cube_non_mo_ens, self.cube_non_mo_det])
+        result = merge_cubes(cubes, 'non_mo_model_config')
+        self.assertIsInstance(result, Cube)
         self.assertArrayAlmostEqual(
-            result.coord("model_id").points, [3000., 4000.])
+            result.coord("model_realization").points, [0., 1., 2., 1000.])
+
+    def test_non_mo_model_id_mismatch(self):
+        """Test that when a model ID attribute string is specified that does
+        not match the model ID attribute key name on both cubes to be merged,
+        an error is thrown"""
+        cubes = iris.cube.CubeList(
+            [self.cube_non_mo_ens, self.cube_non_mo_det])
+
+        # The test cubes contain the 'non_mo_model_config' attribute key.
+        # We'll specify 'non_matching_model_config' as our model ID
+        # attribute key argument. Merge_cubes should then raise a warning
+        # as our specified model ID does not match that on the cubes and it
+        # will not be able to build a model ID coordinate.
+        msg = 'Cannot create model ID coordinate for grid blending ' \
+              'as the model ID attribute specified does not match ' \
+              'that on the cubes to be blended'
+
+        with self.assertRaisesRegex(ValueError, msg):
+            merge_cubes(cubes, 'non_matching_model_config')
```

Comment:
Can we have another test for this error please where one cube has the correct model ID attribute, but one does not?

Reply:
Good idea, I will add one in.

Reply:
Additional test added